### PR TITLE
Feature/fpmm transaction updated

### PR DESCRIPTION
**Pull request by Mi-Lan**
- This is basically astros pr for fpmmTransaction https://github.com/protofire/omen-subgraph/pull/93 updated with master and I included blockNumber to fpmm transaction entity so that I can more easily solve this issue https://github.com/protofire/omen-subgraph/pull/93#issuecomment-733193694 Kaden outlined. Until we find better solution

- I deployed it here https://thegraph.com/explorer/subgraph/mi-lan/omen-rinkeby-fpmm-update

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -253,12 +253,11 @@ type FpmmTransaction @entity {
   user: Account!
   fpmmType: FpmmType!
   transactionType: TransactionType!
-  collateralAmount: BigInt!
-  collateralAmountUSD: BigDecimal!
-  collateralTokenAddress: Bytes!
```
Comment (KadenZipfel) [inline schema.graphql:258]: Do we not need the collateral address for anything?

File: docker-compose.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ services:
       - '8000:8000'
       - '8020:8020'
     environment:
-      postgres_host: 'postgres:5432'
+      postgres_host: 'postgres'
```
Comment (KadenZipfel) [inline docker-compose.yml:25]: What's the purpose of this change?

File: schema.graphql [hunk 3]
```diff
@@ -253,12 +253,11 @@ type FpmmTransaction @entity {
...
   collateralTokenAmount: BigInt!
+  sharesOrPoolTokenAmount: BigInt!
   creationTimestamp: BigInt!
   transactionHash: Bytes!
+  blockNumber:BigInt!
```
Comment (KadenZipfel) [inline schema.graphql:260]: Just out of curiosity, how does including the block number solve the problem of tracking those behind the scenes buys/sells?

File: docker-compose.yml (hunk 2 above)
Comment (Mi-Lan) [reply docker-compose.yml:25]: so that it works for mac users since the way it was it doesn't

File: schema.graphql (hunk 1 above)
Comment (Mi-Lan) [reply schema.graphql:258]: yes for getting token but i can get that out of fpmm so there was no need for it

File: schema.graphql (hunk 3 above)
Comment (Mi-Lan) [reply schema.graphql:260]: So i need to see at what block the transaction was executed and with that blockNumber i query past blockchain state for pool to determine the buy/sell at the time and this saves me one extra step of getting blockNumber from transaction hash instead...

**Comment by KadenZipfel** [comment]
Can you merge master to incorporate this fix: https://github.com/protofire/omen-subgraph/pull/112

**Comment by Mi-Lan** [comment]
> Can you merge master to incorporate this fix: #112

Screenshot 2020-12-28 at 19 48 45
@KadenZipfel as you can see from commits it is incorporated...i don't know why it still has that error?

File: src/FixedProductMarketMakerMapping.ts [hunk 4]
```diff
@@ -132,12 +129,15 @@ function recordFPMMLiquidity(fpmm: FixedProductMarketMaker,
...
-    fpmmTransaction.collateralTokenAmount = fpmmLiquidity.collateralTokenAmount;
+    if (liquidityType === LIQUIDITY_TYPE_ADD) {
+      fpmmTransaction.collateralTokenAmount = max(outcomeTokenAmounts);
+    } else {
+      fpmmTransaction.collateralTokenAmount = collateralRemovedFromFeePool;
+    }
+    fpmmTransaction.sharesOrPoolTokenAmount = sharesAmount;
     fpmmTransaction.creationTimestamp = creationTimestamp;
     fpmmTransaction.transactionHash = transactionHash;
+    fpmmTransaction.blockNumber=blockNumber
```
Comment (KadenZipfel) [inline src/FixedProductMarketMakerMapping.ts:140]: Looks like `blockNumber` isn't defined here maybe?

**Review by KadenZipfel — CHANGES_REQUESTED** [review]

**Comment by Mi-Lan** [comment]
- Here is the published version of this graph https://thegraph.com/explorer/subgraph/mi-lan/omen-graph 
- I deleted blockNumber field which was causing error since that functionality can work without it and separate issue will be created for it

**Review by KadenZipfel — APPROVED** [review]
